compiler/tests/evm_runtime.rs

```diff
@@ -187,6 +187,34 @@ fn test_runtime_sloadn() {
     })
 }
 
+#[test]
+fn test_weird_bug() {
+    with_executor(&|mut executor| {
+        test_runtime_functions(
+            &mut executor,
+            functions::std(),
+            statements! {
+                (let min_val := 0x8000000000000000000000000000000000000000000000000000000000000000)
+                (let val1 := 0x8000000000000000000000000000000000000000000000000000000000000000)
+
+                // Params for i8 that work just fine
+                // (let min_val := 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff80)
+                // (let val1 := 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff80)
+
+                (let val2 := 0x1)
+                // underflow, if val1 < 0, val2 > 0 and val1 < (min_value / val2)
+                //[assert_eq!(0, (and((and((slt(val1, 0)), (sgt(val2, 0)))), (slt(val1, (sdiv(min_val, val2)))))))]
+
+                // i256_min < (i256_min / 1)
+                // e.g. for i8 that would be: -128 < (-128 / 1)
+                // Why does this return 1? rust_evm bug?
+                [assert_eq!(0x1, (slt(val1, (sdiv(min_val, val2)))))]
```

Comment:
@g-r-a-n-t This seems to be the offending part. E.g. if you compute `i8_min < (i8_min / 1)` aka `-128 < (-128 / 1)` the result is `false` / `0x0`. But if you do the same with `i256` it returns `true` / `0x1` which doesn't seem to make sense. I wonder if this is an actual `rust-evm` bug.

I noticed there's a newer rust-evm version released already but nothing suspicious in the changes stood out to me. 

Another idea would be that there's a bug in the conversion from YUL to bytecode but I'm not yet sure how to best validate that.